### PR TITLE
Instance Create: Migrate to egoscale v3 and add multiple sshkeys

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
# Description

## Checklist
(For exoscale contributors)

* [x] Changelog updated (under *Unreleased* block)
* [x] Testing

## Testing

```Bash
go run main.go c i create test-instance \ 
             -z de-fra-1 \
             --anti-affinity-group b2dbf427-251d-404e-83a4-ee804c9fcc04 \
             --cloud-init ~/cloud.yaml \
             --disk-size 100 \
             --instance-type standard.small \
             --ipv6 \
             --label test=toto \
             --private-network fbc1093c-53eb-46cf-a5b6-5cc4c7ac9cc4 \
             --security-group vpn \
             --ssh-key pierro777 \
             --ssh-key test1-1721039696 \
             --ssh-key test1-1721044660 \
             --template "Linux Debian 12 (Bookworm) 64-bit" \
             --template-visibility public

 ✔ Creating instance "test-instance"... 21s
┼──────────────────────┼──────────────────────────────────────┼
│   COMPUTE INSTANCE   │                                      │
┼──────────────────────┼──────────────────────────────────────┼
│ ID                   │ c1926e55-7bb9-4678-84aa-444e994c41bc │
│ Name                 │ test-instance                        │
│ Creation Date        │ 2024-07-24 09:16:24 +0000 UTC        │
│ Instance Type        │ standard.small                       │
│ Template             │ Linux Debian 12 (Bookworm) 64-bit    │
│ Zone                 │ de-fra-1                             │
│ Anti-Affinity Groups │ test                                 │
│ Deploy Target        │ -                                    │
│ Security Groups      │ vpn                                  │
│ Private Instance     │ No                                   │
│ Private Networks     │ test                                 │
│ Elastic IPs          │ n/a                                  │
│ IP Address           │ 89.145.161.94                        │
│ IPv6 Address         │ 2a04:c46:e00:82a1:473:88ff:fe00:12c  │
│ SSH Key              │ pierro777                            │
│ Disk Size            │ 100 GiB                              │
│ State                │ running                              │
│ Labels               │ test:toto                            │
│ Reverse DNS          │                                      │
┼──────────────────────┼──────────────────────────────────────┼
```

I made multiple test like that with all the flags, and private instance also has been tested, since it's slightly different.
I can confirm multiple ssh keys are working (pub keys are present on the instance). To show multiple ssh keys, instance show need to be adapted with egoscale v3 also, but will be done in another PR

### CODE REVIEW

File: cmd/instance_create.go [hunk 1]
```diff
@@ -232,16 +286,22 @@ func (c *instanceCreateCmd) cmdRun(_ *cobra.Command, _ []string) error { //nolin
...
+		}
 	}
 
 	if !globalstate.Quiet {
 		return (&instanceShowCmd{
 			cliCommandSettings: c.cliCommandSettings,
-			Instance:           *instance.ID,
-			Zone:               c.Zone,
+			Instance:           instanceID.String(),
+			// migrate instanceShow to v3 to pass v3.ZoneName
```
Comment (sauterp) [inline cmd/instance_create.go:303]: Is this a TODO?

**Review by sauterp — APPROVED** [review]
That's quite a lot of changes, how did you test it?

File: utils/utils.go [hunk 2]
```diff
@@ -236,3 +237,22 @@ func ForEachZone(zones []string, f func(zone string) error) error {
 
 	return meg.Wait().ErrorOrNil()
 }
+
+// ParseInstanceType returns an v3.InstanceType with family and name.
+func ParseInstanceType(instanceType string) v3.InstanceType {
```
Comment (kobajagi) [inline utils/utils.go:242]: It would be nice to have unit test for this function, it's easy to write.

File: cmd/instance_create.go [hunk 3]
```diff
@@ -41,11 +40,11 @@ type instanceCreateCmd struct {
 	Labels             map[string]string `cli-flag:"label" cli-usage:"instance label (format: key=value)"`
 	PrivateNetworks    []string          `cli-flag:"private-network" cli-usage:"instance Private Network NAME|ID (can be specified multiple times)"`
 	PrivateInstance    bool              `cli-flag:"private-instance" cli-usage:"enable private instance to be created"`
-	SSHKey             string            `cli-flag:"ssh-key" cli-usage:"SSH key to deploy on the instance"`
+	SSHKeys            []string          `cli-flag:"ssh-key" cli-usage:"SSH keys to deploy on the instance"`
```
Comment (kobajagi) [inline cmd/instance_create.go:43]:
```suggestion
	SSHKeys            []string          `cli-flag:"ssh-key" cli-usage:"SSH key to deploy on the instance (can be specified multiple times)"`
```

Comment (kobajagi) [reply cmd/instance_create.go:43]: To be consistent with `PrivateNetworks`.

File: cmd/instance_create.go [hunk 4]
```diff
@@ -75,59 +74,84 @@ func (c *instanceCreateCmd) cmdRun(_ *cobra.Command, _ []string) error { //nolin
...
 	if err != nil {
-		return fmt.Errorf("error retrieving instance type: %w", err)
+		return fmt.Errorf("error listing instance type: %w", err)
+	}
+
+	// c.InstanceType is never empty
+	instanceType := utils.ParseInstanceType(c.InstanceType)
+	for i, it := range instanceTypes.InstanceTypes {
+		if it.Family == instanceType.Family && it.Size == instanceType.Size {
+			instanceReq.InstanceType = &instanceTypes.InstanceTypes[i]
```
Comment (kobajagi) [inline cmd/instance_create.go:139]:
```suggestion
			instanceReq.InstanceType = &instanceTypes.InstanceTypes[i]
			break
```

Comment (pierre-emmanuelJ) [reply cmd/instance_create.go:139]: https://github.com/exoscale/cli/pull/620/commits/1b9d243fba6a71d3787ed799c2b3ffd6283aab4f

File: utils/utils.go (hunk 2 above)
Comment (pierre-emmanuelJ) [reply utils/utils.go:242]: https://github.com/exoscale/cli/pull/620/commits/eeccc70689056d0b68d4414ab845a72e91b91af4

File: cmd/instance_create.go (hunk 1 above)
Comment (pierre-emmanuelJ) [reply cmd/instance_create.go:304]: I'll add it then

**Review by kobajagi — APPROVED** [review]

**Pull request merged**